### PR TITLE
Inject IJW Copy Constructor calls in the JIT instead of in the IL stubs

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Try 2 for https://github.com/dotnet/runtime/pull/106000 after CI decided to not run on it for some reason.

Fixes Issue #100751 and #100033 by injecting copy constructor calls in the JIT instead of injecting them in the IL and then using special hooks/interceptors to handle non-IL-supported/accessible scenarios.

Re-enable GS checks in Reverse P/Invokes now that we handle them correctly.

### CODE REVIEW

**Review by jakobbotsch — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Looks like there's one spot I missed with callis. There's some place that (only when running lots of the tests for some reason) that the modreqs get dropped in the calli signature. Still need to narrow it down.

**Comment by jkoritzinsky** [comment]
Solved the issue, this is ready for review/merge.

File: src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs [hunk 1]
```diff
@@ -360,6 +360,12 @@ internal static int EnumCompareTo<T>(T x, T y) where T : struct, Enum
             return x.CompareTo(y);
         }
 
+
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs:363]:
```suggestion
```

File: src/coreclr/inc/sigparser.h [hunk 2]
```diff
@@ -582,19 +582,32 @@ class SigParser
...
+                if (ELEMENT_TYPE_CMOD_INTERNAL == bElementType)
+                {
+                    void * pMT;
+                    uint8_t required;
+                    if (FAILED(hr = sigTemp.GetByte(&required)))
+                        return hr;
+                    
+                    if (FAILED(hr = sigTemp.GetPointer(&pMT)))
+                        return hr;
+
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/inc/sigparser.h:598]:
```suggestion
```

File: src/coreclr/inc/sigparser.h [hunk 3]
```diff
@@ -643,19 +656,32 @@ class SigParser
...
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/inc/sigparser.h:672]:
```suggestion
```

File: src/coreclr/utilcode/prettyprintsig.cpp [hunk 4]
```diff
@@ -674,6 +674,18 @@ static HRESULT PrettyPrintTypeA(
         sprintf_s(tempBuffer, 64, "pMT: %p", pMT);
         IfFailGo(appendStrA(out, tempBuffer));
         break;
+    
+    case ELEMENT_TYPE_CMOD_INTERNAL:
+        {
+            bool required = *typePtr++ != 0;
+            void* pMT;
+            pMT = *((void* UNALIGNED *)typePtr);
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/utilcode/prettyprintsig.cpp:682]:
```suggestion
            memcpy(&pMT, &typePtr, sizeof(pMT));
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/coreclr/inc/formattype.cpp [hunk 5]
```diff
@@ -694,6 +694,44 @@ PCCOR_SIGNATURE PrettyPrintType(
                 appendStr(out, sz);
                 break;
             }
+            case ELEMENT_TYPE_CMOD_INTERNAL :
+            {
+                // ELEMENT_TYPE_CMOD_INTERNAL <required> <TypeHandle>
```
[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply src/coreclr/inc/formattype.cpp:699]: There's a doc comment on corhdr.h. I'll add more info on the comment to explain the format

**Pull request merged**